### ISSUE TITLE
NPE from Operation.simplify

### ISSUE THREAD

**Issue by btilbrook-nextfaze** [cheptsov/AdvancedExpressionFolding#38]
I'm seeing the following IDE exception when viewing the  `android.support.v7.widget.helper.ItemTouchHelper` source from the Android support lib version `25.3.1`:

IDE is Android Studio 2.4 Preview 3.

```
null
java.lang.NullPointerException
	at com.intellij.advancedExpressionFolding.Operation.simplify(Operation.java:75)
	at com.intellij.advancedExpressionFolding.Expression.simplify(Expression.java:171)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:173)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingBuilder.buildFoldRegions(AdvancedExpressionFoldingBuilder.java:195)
	at com.intellij.lang.folding.LanguageFolding.buildFoldingDescriptors(LanguageFolding.java:83)
	at com.intellij.lang.folding.CompositeFoldingBuilder.buildFoldRegions(CompositeFoldingBuilder.java:60)
	at com.intellij.lang.folding.LanguageFolding.buildFoldingDescriptors(LanguageFolding.java:83)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getFoldingsFor(FoldingUpdate.java:236)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getFoldingsFor(FoldingUpdate.java:222)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getUpdateResult(FoldingUpdate.java:112)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.lambda$updateFoldRegions$1(FoldingUpdate.java:101)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.doCompute(PsiParameterizedCachedValue.java:53)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:65)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.getValue(PsiParameterizedCachedValue.java:43)
	at com.intellij.psi.util.CachedValuesManager.getParameterizedCachedValue(CachedValuesManager.java:97)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.updateFoldRegions(FoldingUpdate.java:97)
	at com.intellij.codeInsight.folding.impl.CodeFoldingManagerImpl.updateFoldRegions(CodeFoldingManagerImpl.java:326)
	at com.intellij.codeInsight.folding.impl.CodeFoldingManagerImpl.updateFoldRegionsAsync(CodeFoldingManagerImpl.java:312)
	at com.intellij.codeInsight.daemon.impl.CodeFoldingPass.doCollectInformation(CodeFoldingPass.java:42)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:70)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:440)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1132)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:433)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:587)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:432)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:408)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:142)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:238)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:406)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:165)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

**Comment by cheptsov** [comment]
Can you try 0.7.5?

**Comment by ciscorucinski** [comment]
I cannot get access to 0.7.5

I was able to update to 0.7.4 but nothing about 0.7.5 is showing up in Android Studio. I uninstalled the plugin but I still can only get access to 0.7.4.

0.7.4 is throwing an exception about a method not existing for me

**Comment by cheptsov** [comment]
This is a known problem of our plugin repository. For some reason if you use the EAP channel, you don't see newer updates from the Stable channel. Try to unregister the EAP channel.

**Comment by ciscorucinski** [comment]
Can you explain how to "unregister the EAP channel"

Actually, I think I found a slight loophole to get it installed via the IntelliJ Plugin window. In `Browse Repositories`, I searched for `fold` and your plugin comes up first. It shows that 0.7.5 is available. I quickly clicked the install button, and it installed 0.7.5.  A second later, the page was updated and only showed that 0.7.4 was available, but going back a page showed that 0.7.5 will be installed after a restart.

Does IntelliJ know about this loophole?

**Comment by btilbrook-nextfaze** [comment]
I just updated to plugin `0.7.5` and AS Preview 5 - works fine now. Thank you!
